### DISCUSSION TITLE
Problem loading @celo/identity WASM in browser

### DISCUSSION THREAD

**Discussion by jsaur** [celo-org/social-connect#245] [Help (Q&A)] [answered]
I'm trying to create a web-app that can interact with ODIS and the identity smart contracts to do phone number registration.  Building on Josh's work there's already a node.js script that does this, and works on the server, here: https://github.com/jsaur/register-number/blob/web-app/index.js

However, when attempting to import OdisUtils from @celo/identity in a web browsers, we run into an issue loading the blind-threshold-bls WASM module: 
https://github.com/jsaur/register-number/blob/web-app/sample-app/pages/index.tsx#L17
The specific error we see is:
```
Error: Cannot find module 'util'
from
node_modules/blind-threshold-bls/src/blind_threshold_bls.js
```
Based on this discord discussion: "A difficulty with this is that the @celo/identity package only includes the WASM builds for use in Node and ReactNative"
https://discord.com/channels/600834479145353243/774302385555243008/953381141228978206
What I'm not sure of is whether this requires a new package release of @celo/identity which would include the blind_threshold_bls WASM for the browser, or if there's some other way in the web-app setup to load a broswer-based WASM module and instruct OdisUtils to use that one.
Here's the full quote from discord for those who don't have access:
"You can certainly call ODIS without the NodeJS script. One way is to, using that script as an example, add all the ODIS calls to your React frontend. A difficulty with this is that the @celo/identity package only includes the WASM builds for use in Node and ReactNative. Using it in the browser requires some work. @joshc previously worked on this, although I am not sure if he found a good way to do it. Here is the underlying library built for the browser https://www.npmjs.com/package/threshold-bls-ffi.

If you alternatively want to make the calls from Django, that will require creating a new SDK. It's certainly possibly if you want to try it. The core cryptography code is implemented in Rust https://github.com/celo-org/celo-threshold-bls-rs and so will need interop to be created"

**Comment by arthurgousset** [comment]
Thanks for posting the details here @jsaur!

**Comment by isabellewei** [comment, marked as answer]
update, a workaround has been found: https://github.com/celo-org/SocialConnect/blob/main/privacy.md#web
